node-repository/src/main/java/com/yahoo/vespa/hosted/provision/restapi/v2/NodePatcher.java

```diff
@@ -113,7 +114,9 @@ private Node applyField(String name, Inspector value) {
             case "flavor" :
                 return node.with(nodeFlavors.getFlavorOrThrow(asString(value)));
             case HARDWARE_FAILURE_TYPE:
-                return node.with(node.status().withHardwareFailure(toHardwareFailureType(asString(value))));
+                Optional<String> hardwareFailure = value.type().equals(Type.NIX) ?
```

Comment:
Consider adding a method on value returning an Optional.of(asString(this)), or empty if NIX.